### PR TITLE
combine factors: fct_cross

### PR DESCRIPTION
**Pull request by tslumley**
Adds fct_cross to address #136 
Creates a new factor containing the combined levels from two or more input factors, similar to base::interaction

### CODE REVIEW

File: R/fct_cross.R [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#' Combine levels from two or more factors to create new factor
```
Comment (hadley) [inline R/fct_cross.R]: "a" new factor?

File: R/fct_cross.R [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+#'
+#' Computes a factor whose levels are all the combinations of the levels of the input factors.
+#'
+#' @param .f A factor or character vector
+#' @param ...  additional factors or character vectors
```
Comment (hadley) [inline R/fct_cross.R]:
```suggestion
#' @param ... Additional factors or character vectors
```

File: tests/testthat/test-fct_cross.R [hunk 3]
```diff
@@ -0,0 +1,36 @@
+context("fct_cross")
+
+test_that("Gives correct levels", {
```
Comment (hadley) [inline tests/testthat/test-fct_cross.R]: I read "as test that gives correct levels", i.e. the label shouldn't start with a capital letter (and this file also needs restyling)

File: R/fct_cross.R [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+#'
+#' @export
+#' @examples
+#' fruit <- factor(c("apple","kiwi","apple","apple"))
+#' colour <- factor(c("green","green","red","green"))
+#' eaten <- c("yes","no","yes","no")
+#' fct_cross(fruit, colour)
+#' fct_cross(fruit, colour, eaten)
+#' fct_cross(fruit, colour, keep_empty=TRUE)
+
```
Comment (hadley) [inline R/fct_cross.R]: Can you please use the tidyverse style? Easiest way is probably to use styler package

File: R/fct_cross.R [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+fct_cross = function(.f,..., sep = ":", keep_empty=FALSE){
+
+    .f <-check_factor(.f)
+
+    flist <- rlang::quos(...)
```
Comment (hadley) [inline R/fct_cross.R]: Why do we need tidyeval here?

**Comment by hadley** [comment]
Basic approach looks sound; just needs a few style tweaks (and I wasn't sure why you needed `quos()`)

File: R/fct_cross.R (hunk 5 above)
Comment (tslumley) [reply R/fct_cross.R]: We don't.  Pass the dots works fine.

File: tests/testthat/test-fct_cross.R (hunk 3 above)
Comment (tslumley) [reply tests/testthat/test-fct_cross.R]: Done

File: R/fct_cross.R (hunk 4 above)
Comment (tslumley) [reply R/fct_cross.R]: Cool. I didn't know about the styler add-in.

**Comment by hadley** [comment]
Looks good!

Can you please add a bullet to NEWS? It should briefly [describe the change](http://style.tidyverse.org/news.html) and end with `(@yourname, #issuenumber)`.

**Comment by hadley** [comment]
Thank you!!

**Comment by ewenharrison** [comment]
Thanks guys, appreciate your work on this.

**Pull request merged**